byway/src/main/webapp/destinations.js

```diff
@@ -0,0 +1,201 @@
+      marker.setMap(null);
+    });
+    markers = [];
+    // For each place, get the icon, name and location.
+    const bounds = new google.maps.LatLngBounds();
+    places.forEach(place => {
+      if (!place.geometry) {
+        console.log("Returned place contains no geometry");
+        return;
+      }
+      const icon = {
+        url: place.icon,
+        size: new google.maps.Size(71, 71),
+        origin: new google.maps.Point(0, 0),
+        anchor: new google.maps.Point(17, 34),
+        scaledSize: new google.maps.Size(25, 25)
+      };
+      // Create a marker for each place.
+      markers.push(
+        new google.maps.Marker({
+          map,
+          icon,
+          title: place.name,
+          position: place.geometry.location
+        })
+      );
+
+      if (place.geometry.viewport) {
+        // Only geocodes have viewport.
+        bounds.union(place.geometry.viewport);
+      } else {
+        bounds.extend(place.geometry.location);
+      }
+    });
+    map.fitBounds(bounds);
+}
+
+/*
+* fetches start location and destinations from DestinationsServlet and adds to DOM
+*/
+function getLocations(){
+  fetchDestinations().then((userLocations) => {
+    document.getElementById('start-location').innerText = "Start Location :" + userLocations.start;
+    const container = document.getElementById('destinations-container');
+    container.innerText = "Destinations:";
+    let destinationArray= userLocations.destinations;
+    destinationArray.forEach((destination) => {
+      let destinationToAdd = document.createElement('p');
+      destinationToAdd.innerText = destination;
+      container.appendChild(destinationToAdd);
+    }) 
+  });
+}
+
+/* 
+* fills Start location Searchbox with previously input
+*/
+function getStartDestination(){
+  fetchDestinations().then((userLocations)=>{
+    if (userLocations.start == null){
+      document.getElementById('start-search-box').value = "";
+    }
+    else{
+      document.getElementById('start-search-box').value = userLocations.start;
+    }
+  });
+}
+
+/* 
+* fetches data from servlet
+*/
+function fetchDestinations(){
+  return fetch('/api/destinations').then(response => response.json());
+}
+
+/* 
+* Gets users current location 
+*/
+function getCurrentAddress(){
+  const geocoder = new google.maps.Geocoder();
+  geocoder.geocode({ 'location': userlatlng}, (results, status) => {
+    if (status === "OK") {
+      if (results[0]) {
+          console.log(results[0].formatted_address);
+        document.getElementById("start-search-box").value=results[0].formatted_address;
+      } else {
+          window.alert("No results found");
+        }
+    } else {
+        window.alert("Geocoder failed due to: " + status);
+      }
+  });
+}
+
+/* 
+* add event listener for submit button
+*/
+window.onload = function(){
+  document.getElementById('user-input-form').addEventListener('submit', (event) => {
+    event.preventDefault();
+    const params = new URLSearchParams();
```

Comment:
For post, it's probably preferable to use FormData over URLSearchParams.

It appears you can also just construct FormData from a form HTML element without having to access the form data elements individually: https://developer.mozilla.org/en-US/docs/Web/API/FormData/Using_FormData_Objects#Retrieving_a_FormData_object_from_an_HTML_form

Reply:
for some reason, jsut replacing URLSearchParams with FormData does not work, I will keep looking into it though

Reply:
got it to work! it is now changed to form data